### ISSUE TITLE
Custom nodes from the network storage pod aren't found

### ISSUE THREAD

**Issue by moteloumka** [blib-la/runpod-worker-comfy#37]
Following the readme file installation tutorial I got the feeling that adding custom nodes through a network storage should so the trick as long as they are in the right folder, however, having ran some requests, I see that any custom nodes are ignored. Sorry to ask here but is there any easy way to reach the extra_model_paths.yaml file with additional paths (I suppose that's the easiest way to add them) or will it be necessary to rebuild a docker container just in order to add a path to the custom nodes. I feel like I'm missing something here, if someone could please indicate how to solve this issue, I would be very thankful!

**Comment by larchman01** [comment]
did you solve this?

**Comment by moteloumka** [comment]
@larchman01 not really, I made a new image, forking this repo and adding the path to the custom nodes in the network storage by adding a line custom_nodes : custom_nodes to the src/extra_model_paths.ymal file. Now comfyUI manages to find the custom nodes but the node themselves seem to have a hard time finding the models that are on the network storage... I imagine that there should be some way to fix this, otherwise the mere existence of this repo wouldn't make too much sense, yet it's currently unclear to me how to make things work.

**Comment by TimPietrusky** [comment]
@moteloumka @larchman01 yes sorry, this was supposed to be the solution, but then we saw that when we do use the custom nodes with a network volume, that they still need to install their dependencies per request. Which is something that is increasing the cold start time. 

In https://github.com/blib-la/runpod-worker-comfy/pull/30 we are working on a solution to bake custom nodes directly into the Docker image. 

Until this is resolved, you need to install them directly into the image manually as described in the README.

**Comment by niko2020** [comment]
#30 is very interesting, but if we have a highly customized ComfyUI installation (on a Runpod Volume), with some scripts that would not be restored by a Manager snapshot, what would be the best practice to bake it into the Docker image?

Cheers for this beautifull project!

**Comment by alka7ex** [comment]
Hi @TimPietrusky @moteloumka 

So i trying to build a docker custom image also due to i need to add custom node, but the questions is do you guys build the images locally or using github actions?

im trying to build the docker images using github actions but facing an issues where the runner run out of disk is there any alternatives?

```System.IO.IOException: No space left on device : '/home/runner/runners/2.319.1/_diag/Worker_20240907-125323-utc.log'```

**Comment by moteloumka** [comment]
@alka7ex I found that the easiest way was installing docker and building the thing locally then pushing it on docker hub. It's a bit tricky bc you'll have to install docker and have good internet connection (otherwise downloading all the packages takes ages as much as uploading the whole thing into docker hub) but overall, if you figure out how to use the basics of docker, you're sure to have full control on your custom image and use this repo as a great starting spot that takes care of all the functionalities.

**Comment by drommerkiller** [comment]
> @alka7ex I found that the easiest way was installing docker and building the thing locally then pushing it on docker hub

Is there a guide on building a comfyui docker image that will work with this repo? As you said, it takes time so so a guide how to do it would be great. There is a guide how to create stable diffusion runpod image, but comfy differs enough that it did not work when i tested.

Custom nodes and the dependencies they need are quite essential for ComfyUI. I'm looking alternative for Replicate as there the coldboots/queue has gone off the roof for custom comfyui model. I'm talking 5-15min waiting for running 30sec inference.

**Comment by TimPietruskyRunPod** [comment]
Custom nodes are intended to be **baked into the Docker image at build time** via \`comfy-node-install\`, not loaded from a network volume at runtime. This is intentional: custom nodes often have Python dependencies that need to be installed system-wide, and many custom nodes don't tolerate being added or removed without an interpreter restart. Network volumes are for *data* (models, LoRAs, embeddings, workflow outputs) — not for code.

If you really need a custom node and it's not in your image, derive a new image:

\`\`\`dockerfile
FROM runpod/worker-comfyui:5.8.5-base
RUN comfy-node-install your-custom-node
\`\`\`

…and use that image for your endpoint.

Closing as the supported flow is the derived-Dockerfile pattern.
